lib/chef/http/authenticator.rb

```diff
@@ -83,8 +85,54 @@ def client_name
         @auth_credentials.client_name
       end
 
+      def detect_certificate_key(client_name)
+        self.class.detect_certificate_key(client_name)
+      end
+
+      def check_certstore_for_key(client_name)
+        self.class.check_certstore_for_key(client_name)
+      end
+
+      def retrieve_certificate_key(client_name)
+        self.class.retrieve_certificate_key(client_name)
+      end
+
+      # Detects if a private key exists in a certificate repository like Keychain (macOS) or Certificate Store (Windows)
+      #
+      # @param client_name - we're using the node name to store and retrieve any keys
+      # Returns true if a key is found, false if not. False will trigger a registration event which will lead to a certificate based key being created
+      #
+      #
+      def self.detect_certificate_key(client_name)
+        if ChefUtils.windows?
+          check_certstore_for_key(client_name)
+        else # generic return for Mac and LInux clients
+          false
+        end
+      end
+
+      def self.check_certstore_for_key(client_name)
+        powershell_code = <<~CODE
+          $cert = Get-ChildItem -path cert:\\LocalMachine\\My -Recurse -Force  | Where-Object { $_.Subject -Match "#{client_name}" } -ErrorAction Stop
```

Comment:
We're not checking for chef-#{client_name} here, but we're looking for that here https://github.com/chef/chef/pull/11834/files#diff-b13537c03b46d265d88b542373ce52b7f8ba190f1f60c2f99ee7ab91d15751d3R644

We should make this consistent and ideally on set the chef- prefix in one place so it' snot something we have to track (eg, at the point we're making the function call to detect_certificate_key.

Reply:
I added a method that creates and returns a label of win_cert_client and am using that consistently throughout the authenticator code